app/jobs/regular/process_post.rb

```diff
@@ -33,7 +33,14 @@ def execute(args)
           # TODO stop/restart the worker if needed, let's gather a few here first
           Rails.logger.warn("Cooked post processor in FATAL state, bypassing. You need to urgently restart sidekiq\norig: #{orig_cooked}\nrecooked: #{recooked}\ncooked: #{cooked}\npost id: #{post.id}")
         else
-          post.update_column(:cooked, cp.html)
+          # Hack to prevent race conditions. If a post is cooked, edited and
+          # quickly rebaked, there is a chance that the first rebake will
+          # finish last. This leads to inconsistencies where the 'cooked'
+          # column contains the old baked 'raw' text.
+          Post
+            .where(id: post.id, raw: post.raw)
+            .update_all(cooked: cp.html)
+          post.reload
```

Comment:
`post.reload` will `SELECT` it again from the database. Could we instead do `post.cooked = cp.html` since we know that's the only attribute that changed? You might also need to clear the dirty active record attributes after you do that because we know the value has been saved.

Reply:
That is how I wanted to implement this and it seemed safer that way. Having another look now, it really looks better. 😊